patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1456,7 +1461,7 @@
 +    @SideOnly (Side.CLIENT)
 +    public Vec3d getFogColor(World world, BlockPos pos, IBlockState state, Entity entity, Vec3d originalColor, float partialTicks)
 +    {
-+        if (state.func_185904_a() == Material.field_151586_h)
++        if (state.func_185904_a().isLiquidOtherThanLava())
```

Comment:
It's out-of-scope for this PR, but this is the place I really, really, really want to be replaced with `state.func_185904_a().getFogColor()`...

Reply:
What do you mean? The fog color when you're inside a fluid already changes to the fluid color. I submitted a separate PR for this recently.

Reply:
See #4462